Documentation/Testing/ExtensionTesting.rst

```diff
@@ -89,20 +81,20 @@ Testing enetcache
 
 The extension `enetcache <https://github.com/lolli42/enetcache>`_ is a small extension that helps
 with frontend plugin based caches. It has been available as composer package and a TER extension for quite
-some time and is loosely maintained to keep up with current Core versions. At the time of
-writing, it has three branches:
+some time and is loosely maintained to keep up with current Core versions. At the
+time this page was originally written, it had three branches:
```

Comment:
I don't like "at the time this page was originally written". How should the average reader know when this was? Last month? Last year? Three years ago? I suggest to add a time like "April 2023". This way it is clear that the example may be outdated or not.

Reply:
The point here - the versions are not relevant. It is about the general principle. Perhaps we should use made up versions or something instead.

Alternatively we can update the versions here as well to reflect the current versions of enetcache (but I would have to check if that is still relevant for newer versions) and it is (more) difficult to maintain than just replacing the versions - you have to check which branches of enetcache support which TYPO3 versions.

Reply:
State a date (e.g. today) and write it which is correct on that that. And add a hint with a link to repo that the actual list of branches and version can be looked up in the source repository (aka source of truth)

Reply:
What was originally written, (in v11 branch before changes) does not comply with current main branch.

In previous version of this docs page, the master branch of enetcache was compatible with v9, each branch supported only **one** major TYPO3 version.

But now we have situation that main of enetcache supports 2 versions:


`"typo3/cms-core": "^11.0 || ^12.0"`

And it tests 2 versions.

But in several places of the "Extension Testing" page we still assume only one TYPO3 version is tested and only one can be tested, e.g.

>  This documentation assumes an extension is tested with only one major Core version.

We even discourage to test with 2 core versions with typo3 testing-framework (and recommend nimut):

> If you need test setups for an extension that supports
  multiple major Core versions at the same time, you may run into trouble if using
  the `typo3/testing-framework <https://github.com/TYPO3/testing-framework>`_ package


So, I think we should also decide, what do we document? Testing 2 TYPO3 versions in 1 branch or 1 TYPO3 version only in one branch?

For the former, enetcache would be better example, for the latter, styleguide might be a better example.

Also, I am not sure if this still applies:

> The Core Team encourages extension developers to have dedicated Core branches  per Core version.

-----

**Conclusion:**


For now I will try to come to an end with this PR and base it on "enetcache" but still test only one core version. I do not want to make too many changes at once because it still should be up-ported to main and v12. 

It is not a complete overhaul of the "Extension Testing" page but hopefully some outdated stuff is removed.

I add a note at the top what is still unclear or missing: this is for the readers and contributors and maintainers.